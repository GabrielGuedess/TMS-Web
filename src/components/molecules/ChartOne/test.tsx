import { type ReactNode } from 'react';

import { render } from '@testing-library/react';

import { ChartOne } from '.';

jest.mock('components/atoms/ApexChart', () => ({
  ApexChart: ({ children }: { children: ReactNode }) => (
    <div data-testid="Mock ApexChart">{children}</div>
  ),
}));

describe('<ChartOne />', () => {
  it('should render the chart one correctly', () => {
    const { container } = render(<ChartOne />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
