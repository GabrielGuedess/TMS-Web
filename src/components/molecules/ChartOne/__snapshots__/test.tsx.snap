// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartOne /> should render the chart one correctly 1`] = `
<div
  class="flex w-full flex-col gap-4 rounded border border-zumthor-100 bg-white-lilac-50 px-5 pb-5 pt-[1.875rem] shadow-default transition-all dark:border-shark-950 dark:bg-smoke-950 sm:px-[1.875rem] md:grid-cols-2 md:gap-6 xl:col-span-8"
>
  <div
    class="flex flex-wrap items-start justify-between gap-3 sm:flex-nowrap"
  >
    <div
      class="flex w-full flex-wrap gap-3 sm:gap-5"
    >
      <div
        class="flex gap-2"
      >
        <div
          class="mt-1 flex size-4 items-center justify-center rounded-full border border-cobalt-900"
        >
          <div
            class="flex size-2 rounded-full bg-cobalt-900"
          />
        </div>
        <div
          class="w-full"
        >
          <p
            class="font-semibold text-cobalt-900"
          >
            Total Revenue
          </p>
          <p
            class="text-sm font-medium text-comet-500 dark:text-dark-300"
          >
            12.04.2022 - 12.05.2022
          </p>
        </div>
      </div>
      <div
        class="flex gap-2"
      >
        <div
          class="mt-1 flex size-4 items-center justify-center rounded-full border border-red-500"
        >
          <div
            class="flex size-2 rounded-full bg-red-500"
          />
        </div>
        <div
          class="w-full"
        >
          <p
            class="font-semibold text-primary-400"
          >
            Total Sales
          </p>
          <p
            class="text-sm font-medium text-comet-500 dark:text-dark-300"
          >
            12.04.2022 - 12.05.2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="mt-8 flex w-full justify-center sm:mt-0 sm:justify-end"
    >
      <div
        class="inline-flex items-center gap-2 rounded-md bg-porcelain-50 p-1.5 transition-all dark:bg-shark-950"
      >
        <button
          class="rounded bg-white-lilac-50 px-3 py-1 text-xs font-medium text-comet-500 shadow-card outline-primary-400 transition-all hover:bg-white-lilac-50 hover:shadow-card dark:bg-smoke-950 dark:text-white-lilac-50 dark:hover:bg-smoke-950"
          type="button"
        >
          Day
        </button>
        <button
          class="rounded px-3 py-1 text-xs font-medium text-comet-500 outline-primary-400 transition-all hover:bg-white-lilac-50 hover:shadow-card dark:text-white-lilac-50 dark:hover:bg-smoke-950"
          type="button"
        >
          Week
        </button>
        <button
          class="rounded px-3 py-1 text-xs font-medium text-comet-500 outline-primary-400 transition-all hover:bg-white-lilac-50 hover:shadow-card dark:text-white-lilac-50 dark:hover:bg-smoke-950"
          type="button"
        >
          Month
        </button>
      </div>
    </div>
  </div>
  <div
    class="h-[355px] w-full"
  >
    <div
      data-testid="Mock ApexChart"
    />
  </div>
</div>
`;
